### PR TITLE
New visual basic cheatsheet : continuing #2131

### PR DESCRIPTION
**Pull request by gautamkrishnar**
added visual basic cheat sheet
#2131

---------------------------
Ia page: https://duck.co/ia/view/visual_basic_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=visual basic cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/visual_basic_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=visual basic cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/visual_basic_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=visual basic cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/visual_basic_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/visual-basic.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
+{
+    "id":"visual_studio_cheat_sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/visual-basic.json]: @gautamkrishnar looks like the filename and ID don't match -- which is correct?

Seems like the ID matches the IA page so the filename should probably change

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=visual basic cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/visual_basic_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/visual-basic.json (hunk 1 above)
Comment (mohan08p) [inline share/goodie/cheat_sheets/json/visual-basic.json]: @gautamkrishnar : change the file name from **visual-basic.json** to **visual-studio.json**

**Comment by gautamkrishnar** [comment]
@moollaza done :+1: 
shall i have to change anything more....?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=visual basic cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/visual_basic_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=visual basic cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/visual_basic_cheat_sheet_mobile.png)

**Comment by mintsoft** [comment]
@gautamkrishnar This PR has conflicts on it now; can you rebase & resolve?

**Comment by gautamkrishnar** [comment]
@mintsoft Fixed ;)

File: share/goodie/cheat_sheets/json/visual-studio.json [hunk 2]
```diff
@@ -1,5 +1,85 @@
 {
     "id":"visual_studio_cheat_sheet",
+<<<<<<< HEAD
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/visual-studio.json]: @gautamkrishnar Looks like we've got... merge conflicts? A whole new meaning...

**Comment by GuiltyDolphin** [comment]
@gautamkrishnar Looks like we've already got a [visual studio cheat sheet](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/share/goodie/cheat_sheets/json/visual-studio.json)? Would you rather update the aliases of that one?

File: share/goodie/cheat_sheets/json/visual-studio.json (hunk 2 above)
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/visual-studio.json]: woops! will fix that. Btw @GuiltyDolphin  this is actually a visual basic cheat sheet, not a visual studio cheatsheet, sorry for the confusion. i will update it

**Comment by gautamkrishnar** [comment]
:heart: fixed it! :white_check_mark:

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=visual basic cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/visual_basic_cheat_sheet_mobile.png?layhREAqcq)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=visual basic cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/visual_basic_cheat_sheet_mobile.png?O1X2vYPVrJ)

File: share/goodie/cheat_sheets/json/visual-basic.json [hunk 3]
```diff
@@ -0,0 +1,82 @@
+{
+    "id":"visual_basic_cheat_sheet",
+    "name":"Visual basic",
+    "description": "Visual basic is an integrated development environment (IDE) from to develop in visual basic language",
+    "metadata":{
+        "sourceName":"techrepublic",
+        "sourceUrl":"http://www.techrepublic.com/blog/microsoft-office/37-keyboard-shortcuts-for-moving-faster-in-the-visual-basic-editor/"
+    },
+  "aliases": [
+    "visual basic"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/visual-basic.json]: this is causing the test to fail -- please remove as this is unnecessary.

**Comment by gautamkrishnar** [comment]
Okay closing this one...  :+1:

**Pull request closed without merging**